core/src/main/java/io/aiven/kafka/tieredstorage/fetch/cache/DiskChunkCache.java

```diff
@@ -98,11 +98,16 @@ public RemovalListener<ChunkKey, Path> removalListener() {
         return (key, path, cause) -> {
             try {
                 if (path != null) {
-                    final long fileSize = Files.size(path);
-                    Files.delete(path);
-                    metrics.chunkDeleted(fileSize);
-                    log.trace("Deleted cached file for key {} with path {} from cache directory."
-                        + " The reason of the deletion is {}", key, path, cause);
+                    if (Files.exists(path)) {
+                        final long fileSize = Files.size(path);
+                        Files.delete(path);
```

Comment:
Considering the _racy_ nature of things here, a file may be deleted between the `Files.exists` check and `Files.delete`. I think, we should instead try to delete without a check and catch the "file not found" exception. "It is better to ask forgiveness than permission" :)